src/Features/ExtendedTranslucency.cpp

```diff
@@ -0,0 +1,160 @@
+#include "ExtendedTranslucency.h"
+
+#include "../Util.h"
+
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT(
+	ExtendedTranslucency::MaterialParams,
+	AlphaMode,
+	AlphaReduction,
+	AlphaSoftness);
+
+const RE::BSFixedString ExtendedTranslucency::NiExtraDataName = "AnisotropicAlphaMaterial";
+
+ExtendedTranslucency* ExtendedTranslucency::GetSingleton()
+{
+	static ExtendedTranslucency singleton;
+	return &singleton;
+}
+
+void ExtendedTranslucency::SetupResources()
+{
+	// Default material model buffer only changes in settings UI
+	materialDisableCB.emplace(ConstantBufferDesc<MaterialParams>());
+	MaterialParams disabled{ 0, 0.f, 0.f, 0 };
+	materialDisableCB->Update(disabled);
+	// Default material model buffer only changes in settings UI
+	materialDefaultCB.emplace(ConstantBufferDesc<MaterialParams>());
+	materialDefaultCB->Update(settings);
+	// Dynamic material model buffer changes when we read materials
+	materialDynamicCB.emplace(ConstantBufferDesc<MaterialParams>(true));
+}
+
+void ExtendedTranslucency::BSLightingShader_SetupGeometry(RE::BSRenderPass* pass)
+{
+	auto renderer = RE::BSGraphics::Renderer::GetSingleton();
+	auto* transcluency = ExtendedTranslucency::GetSingleton();
+
+	ID3D11DeviceContext* context = reinterpret_cast<ID3D11DeviceContext*>(renderer->GetRuntimeData().context);
+	ID3D11Buffer* buffers[1];
+	if (auto* data = pass->geometry->GetExtraData(NiExtraDataName)) {
```

Comment:
Not sure how good for perf it to look for ExtraData in each PerGeometry

Reply:
I think it should be fine compared to the cost of updating constant buffer:
The ExtraData is empty on most geometry, and for others with a small amount of extra data, it's a linear/binary lookup on an array of char*, the compare is pointer based instead of strcmp.

I tried to reuse some shader flags, like VertexAlpha, but found there are lot of random assets using them 😂